src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -293,12 +298,7 @@ private static string[] GetAllAvailableSDKs()
             string[]? resolvedPaths = null;
             foreach (string dotnetPath in s_dotnetPathCandidates.Value)
             {
-                int rc = NativeMethods.hostfxr_get_available_sdks(exe_dir: dotnetPath, result: (key, value) => resolvedPaths = value);
-
-                if (rc == 0 && resolvedPaths != null && resolvedPaths.Length > 0)
-                {
-                    break;
-                }
+                NativeMethods.hostfxr_get_available_sdks(exe_dir: dotnetPath, result: (key, value) => resolvedPaths = value);
```

Comment:
How about this?

```suggestion
                int rc = NativeMethods.hostfxr_get_available_sdks(exe_dir: dotnetPath, result: (key, value) => resolvedPaths = value);

                if (rc == 0 && resolvedPaths != null && resolvedPaths.Length > 0)
                {
                    foreach (string path in resolvedPaths)
                    {
                        yield return path;
                    }
                }
```

Reply:
Ah, thanks in my head it was accumulating them but I see now it isn't. Will update.

Reply:
Updated.

Reply:
Definitely a bit confusing with how the native APIs work 🙂 I struggled with that for a bit when I wrote the first version of this.